apollo-federation-types/src/build_plugin/plugin_result.rs

```diff
@@ -11,12 +11,42 @@ pub struct PluginResult {
     pub schema: Option<String>,
     pub build_messages: Vec<BuildMessage>,
 
+    // We need to keep the default for backwards compatibility
+    #[serde(default = "bool::default")]
+    is_non_build_failure: bool,
+
     /// Other untyped JSON included in the build output.
     #[serde(flatten)]
     other: crate::UncaughtJson,
 }
 
 impl PluginResult {
+    /// Returns true if the failure is due to something other than the execution of the underlying javascript code
+    /// e.g. the supergraph yaml is misconfigured or the json recieved is malformed
+    pub fn is_non_build_failure(&self) -> bool {
+        self.is_non_build_failure && self.schema.is_none() && !self.is_success
+    }
```

Comment:
we kinda have this concept in this repo w/the `BuildErrorType` enum:

```
#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
#[non_exhaustive]
pub enum BuildErrorType {
    Composition,
    Config,
}
```

- i like modeling things with enums specifically like this just to make things a bit easier to read. if you stored this type on the struct, this function becomes

```suggestion
    pub fn is_non_build_failure(&self) -> bool {
        !matches!(self, BuildErrorType::Composition)
    }
```

Reply:
Went with an enum, also using a result to make it more clear we will have a success or error reason. I think it makes the data object a bit easier to understand.